### DISCUSSION TITLE
Tag specified out of required order

### DISCUSSION THREAD

**Discussion by szistvan** [quickfix-j/quickfixj#571] [Q&A]
Hi,
I am sending 35=V MDRequest messages to an acceptor, where the 146 repeating group precedes the 267 group. Which is normal. Hovewer the specification of the acceptor specifies a different order. And I always get rejection with "Tag specified out of required order" message. Since I can not make the acceptor more adaptable I need to change the order of the fields in my side. But they are not in a group, where it is "easy", they are the fields/groups of the main message. As far as I seen the orders can not be modified, as the fields and groups are stored in a TreeMap. Dictionary modifications don't have any effects. The only solution is really a dictionary modification and then a full build and using the newly built jar?

**Comment by chrjohn** [comment]
What do you mean by "they are the fields/groups of the main message"? Outside of repeating groups the order should not matter. 

You could try if this helps: https://stackoverflow.com/a/60048848/4962355

**Reply by szistvan** [reply]
The reject is about a field and not a repeating group. There are two QuickFixJ settings, ValidateUnorderedGroupFields and ValidateFieldsOutOfOrder, they suggest that not only the group fieds could be checked. 
Otherwise I solved the problem. The Session.class uses the Message.toString() for creating the message to send. I overwrote the toString method of the MDRequest class, change some fields by String manipulation. It can work as it is a rare message, no performance issue.

**Reply by chrjohn** [reply]
> Message.toString() for creating the message to send. I overwrote the toString method of the MDRequest

Well, that's another way to do it. Didn't hear of that before but as long as it works. :)
